test.py

```diff
@@ -342,6 +342,38 @@ def test_set_get_policy_aes_128_cbc(directory):
                      filenames="AES-128-CTS")
 
 
+def test_set_get_policy_sm4(directory):
+    """Tests getting and setting an encryption policy that uses SM4-XTS
+    contents encryption and SM4-CTS filenames encryption.  Note that the
+    kernel doesn't yet support SM4-XTS contents encryption and SM4-CTS
+    filenames encryption, so that is not tested."""
+
+    # Skip the test if the kernel lacks support for SM4-XTS and SM4-CTS.
+    try:
+        prepare_encrypted_dir(directory, "--contents=SM4-XTS",
+                              "--filenames=SM4-CTS")
+    except SystemError as e:
+        # Old kernel that doesn't know about SM4-XTS and SM4-CTS
+        assert "invalid encryption options provided" in str(e)
+        pytest.skip("Kernel doesn't support SM4-XTS and SM4-CTS encryption")
+    except OSError as e:
+        assert "Package not installed" in str(e)
+        pytest.skip("Kernel doesn't support SM4-XTS and SM4-CTS encryption")
+
+    for padding in [4, 16, 32, None]:
```

Comment:
Maybe leave out the non-default filename padding tests, for simplicity.  Non-default filenames padding is already tested with other modes.  Testing it again with SM4 adds very little value, IMO.